core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -753,6 +755,38 @@ public RexNode convertWindowFunction(
     return cx.getRexBuilder().makeCall(returnType, fun, exprs);
   }
 
+  public RexNode convertSessionWindowFunction(
+      SqlRexContext cx,
+      SqlSessionTableFunction fun,
+      SqlCall call) {
+    // The first operand of window function is actually a query, skip that.
+    final List<SqlNode> operands = Util.skip(call.getOperandList());
+    final List<RexNode> exprs =
+        convertOperands(cx, call, operands,
+            SqlOperandTypeChecker.Consistency.NONE);
+    RelDataType returnType =
```

Comment:
No need to support DESCRIPTOR(partition_key), it is more verbose than standard and hard to use.

Reply:
Hi, Danny, drop the old DESCRIPTOR(partition_key) syntax is an option.
however it would cause compatibility problem, I'm not sure whether the user could accept this uncompatible update.